feat: expose deployment to serve Executors (#5563)
jina-ai/jina: docs/concepts/executor/serve.md

(serve-executor-standalone)=
# Serve

{class}`~jina.Executor`s can be served and accessed over gRPC, allowing you to use them to create a gRPC-based service for various tasks such as model inference, data processing, generative AI, and search services.

There are different options for deploying and running a standalone Executor:
* Run the Executor directly from Python with the {class}`~jina.orchestrate.deployments.Deployment` class

> * Run the static {meth}`~jina.serve.executors.BaseExecutor.to_kubernetes_yaml()` method to generate K8s deployment configuration files

Reviewer: ```suggestion
* Run the static {meth}`~jina.serve.executors.BaseExecutor.to_kubernetes_yaml()` method to generate Kubernetes deployment configuration files
```

````

## Serve directly
An {class}`~jina.Executor` can be served using the {class}`~jina.orchestrate.deployments.Deployment` class.

The {class}`~jina.orchestrate.deployments.Deployment` class aims at separating the deployment configuration from the 

> serving logic. 

Reviewer: ```suggestion
The {class}`~jina.orchestrate.deployments.Deployment` class aims to separate the deployment configuration from the serving logic.

In other words:
* the Executor cares about defining the logic to serve, which endpoints to define and what data to accept.

> * the Deployment layer cares about how to orchestrate this service, how many replicas or shards,...

Reviewer: ```suggestion
* the Deployment layer cares about how to orchestrate this service, how many replicas or shards, etc.
```

> This separation also aims at enhancing the reusability of Executors: the same implementation of an Executor can be 

Reviewer: ```suggestion
This separation also aims to enhance the reusability of Executors: the same implementation of an Executor can be 
```

````

> (reload-executor)=

Author: ```suggestion
```

````

(reload-executor)=

> ## Reload Executor

Reviewer 2: why is this removed?

Author: content duplicated here https://feat-deployment-serve--jina-docs.netlify.app/concepts/executor/hot-reload/

Author: removed
